src/libYARP_math/src/RandScalar.cpp

```diff
@@ -13,29 +13,28 @@
 #include <cstdio>
 #include <cmath>
 
-// implementation of Marsenne Twister
-#include <yarp/math/impl/mt.h>
+// implementation of Marsenne Twister from C++11's library
+#include <random>
 
 using namespace yarp::sig;
 using namespace yarp::math;
 
-inline MersenneTwister *implementation(void *t)
+inline std::mt19937 *implementation(void *t)
```

Comment:
Since you are cleaning this, can we remove this method and instead just use the `static_cast` where it is called?

Reply:
Actually it is ok, just leave it like this for now...

The actual thing that should be done is to have just one pointer that is a class that contains both the `std::mt19937` and the seed, in order to remove all private member, except for the opaque pointer, from the interface